extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/util/MultiStageQueryContext.java

```diff
@@ -87,6 +90,8 @@ public class MultiStageQueryContext
 
   public static final String CTX_DURABLE_SHUFFLE_STORAGE = "durableShuffleStorage";
   private static final boolean DEFAULT_DURABLE_SHUFFLE_STORAGE = false;
+  public static final String CTX_LIMIT_SELECT_RESULT = "limitSelectResult";
```

Comment:
Lets rename this to `selectDestination="report"` and then not limit the results to maintain backward compatibility.  
I am working on another PR which can use this limiting where` selectDestination="Report+s3"`.

Reply:
Changed this. I've also added an S3 select destination, so that I can add a test to see if it truncates, but have left it a bit light on javadocs. This can be added depending on the implementation details of your PR.